datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -1240,6 +1278,35 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn or_test() -> Result<()> {
+        // check that (lhs OR rhs) is equal to result
+        // ((lhs_lower, lhs_upper), (rhs_lower, rhs_upper), (lower_result, upper_result))
+        let cases = vec![
+            // Note there are only three valid boolean intervals: [false, false], [false, true] and [true, true],
+            ((false, false), (false, false), (false, false)),
```

Comment:
This test might be easier to read if I rewrote it in terms of `CERTAINLY_TRUE`, `UNCERTAIN, and `CERTAINLY_FALSE`. I'll try and do that next week